tests/conftest.py

```diff
@@ -327,31 +330,17 @@ def get_logger() -> GetLogger:
 
 
 @pytest.fixture()
-async def redis_client(docker_ip: str, redis_service: None) -> AsyncGenerator[AsyncRedis, None]:
-    # this is to get around some weirdness with pytest-asyncio and redis interaction
-    # on 3.8 and 3.9
-
-    Redis(host=docker_ip, port=6397).flushall()
-    client: AsyncRedis = AsyncRedis(host=docker_ip, port=6397)
-    yield client
-    try:
-        await client.aclose()  # type: ignore[attr-defined]
-    except RuntimeError:
-        pass
+async def redis_client(redis_service: None, docker_ip: str):
+    async with Redis(host=docker_ip, port=6397) as client:
+        yield client
+        await client.flushall()
```

Comment:
overall this feels more natural and the try / except of the original fixture is just a way to mask unclosed ressources